spyder/preferences/shortcuts.py

```diff
@@ -122,47 +156,77 @@ def keyPressEvent(self, event):
         super(ShortcutEditor, self).__init__(parent)
         self._parent = parent
 
         self.context = context
-        self.npressed = 0
-        self.keys = set()
-        self.key_modifiers = set()
-        self.key_non_modifiers = list()
-        self.key_text = list()
-        self.sequence = sequence
-        self.new_sequence = None
-        self.edit_state = True
+        self.name = name
         self.shortcuts = shortcuts
+        self.cur_sequence = sequence
 
+        self.new_subseq = list()
+        self.new_sequence = ', '.join(self.new_subseq)
+        self.new_qsequence = QKeySequence(self.new_sequence)
+
+        self.setup()
+
+    def setup(self):
+        """Setup the ShortcutEditor with the provided arguments."""
         # Widgets
         self.label_info = QLabel()
-        self.label_info.setText(_("Press the new shortcut and select 'Ok': \n"
-             "(Press 'Tab' once to switch focus between the shortcut entry \n"
-             "and the buttons below it)"))
+        self.label_info.setText(
+            _("Press the new shortcut and select 'Ok' "
+              "or press 'Clear' to unbind the shortcut."))
+        self.label_info.setWordWrap(True)
+
         self.label_current_sequence = QLabel(_("Current shortcut:"))
-        self.text_current_sequence = QLabel(sequence)
+        self.text_current_sequence = QLabel(self.cur_sequence)
+
         self.label_new_sequence = QLabel(_("New shortcut:"))
-        self.text_new_sequence = CustomLineEdit(self)
-        self.text_new_sequence.setPlaceholderText(sequence)
+        self.text_new_sequence = ShortcutLineEdit(self)
+        self.text_new_sequence.setPlaceholderText(_("Press shortcut."))
+
         self.helper_button = HelperToolButton()
-        self.helper_button.hide()
+        self.helper_button.setIcon(QIcon())
         self.label_warning = QLabel()
-        self.label_warning.hide()
+        self.label_warning.setWordWrap(True)
+        self.label_warning.setAlignment(Qt.AlignTop | Qt.AlignLeft)
+        self.label_warning.setMinimumHeight(
+            3 * self.label_warning.sizeHint().height())
 
-        bbox = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
-        self.button_ok = bbox.button(QDialogButtonBox.Ok)
-        self.button_cancel = bbox.button(QDialogButtonBox.Cancel)
+        self.button_default = QPushButton(_('Default'))
+        self.button_ok = QPushButton(_('Ok'))
+        self.button_ok.setEnabled(False)
+        self.button_clear = QPushButton(_('Clear'))
+        self.button_cancel = QPushButton(_('Cancel'))
+        button_box = QHBoxLayout()
+        button_box.addWidget(self.button_default)
+        button_box.addStretch(100)
+        button_box.addWidget(self.button_ok)
+        button_box.addWidget(self.button_clear)
+        button_box.addWidget(self.button_cancel)
+
+        # New Sequence button box
+        self.btn_clear_sequence = create_toolbutton(

```

Comment:
Since it clears the entire dialog vs. just deleting the most recent shortcut, it stands to reason that this should go second, and the "delete last shortcut" button first (i.e. swap them).